core/src/main/kotlin/com/strumenta/kolasu/parsing/Parsing.kt

```diff
@@ -176,26 +185,28 @@ abstract class KolasuParser<R : Node, P : Parser, C : ParserRuleContext> {
 
     fun lex(inputStream: InputStream, onlyFromDefaultChannel: Boolean = true): LexingResult {
         val issues = LinkedList<Issue>()
-        val lexer = createANTLRLexer(inputStream)
-        lexer.removeErrorListeners()
-        lexer.injectErrorCollectorInLexer(issues)
         val tokens = LinkedList<Token>()
-        do {
-            val t = lexer.nextToken()
-            if (t == null) {
-                break
-            } else {
-                if (!onlyFromDefaultChannel || t.channel == Token.DEFAULT_CHANNEL) {
-                    tokens.add(t)
+        val time = measureTimeMillis {
```

Comment:
Does measuring slow us down potentially?

Reply:
I highly doubt it. My only concern is that it's not available for all platforms.
They have experimental cross-platform time APIs but... they're experimental and subject to change.

Reply:
To elaborate, measureTimeMillis only does a system.currentTimeMillis() at the start and one at the end of the block and returns the difference. It's not instrumenting the code or other invasive stuff.